### PR TITLE
feat: Test for Prisma Client auto-install on generate #19259

### PR DESCRIPTION
**Pull request by Templer44**
Tests created for issues #19259 and #19814 to check for [object promise] in error message on generate when using yarn in a Windows environment.

Also implemented code to describe [object Promise] if displayed in error message.

### CODE REVIEW

File: packages/client/tests/e2e/prisma-client-error-message-win32/readme.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#Prisma Client Test for Windows Environment
+
+```
+Tests were created to check if the prisma client is installed properly for the Windows environment.
+For the Windows environments, the prisma client is unable to resolve packages and paths automatically compared to other environments when using Yarn. This results in a "Could not resolve @prisma/client..." error. Meaning that the client is unable to find the @prisma/client or resolve the path to or packages for the @prisma/client.
+
+Thus, Windows users must manually check if both the packages and @prisma/client are installed properly in order to successfully run the client.
+```
```
Comment (Jolg42) [inline packages/client/tests/e2e/prisma-client-error-message-win32/readme.md:8]:
```suggestion
# Prisma Client Test on Windows

```
Tests were created to check if Prisma Client is installed properly on Windows.
On Windows, Prisma Client is unable to resolve packages and paths automatically compared to other environments when using Yarn. This results in a "Could not resolve @prisma/client..." error. Meaning that the client is unable to find the `@prisma/client` or resolve the path to or packages for `@prisma/client`.

Thus, Windows users must manually check if both the packages and `@prisma/client` are installed properly in order to successfully run the client.
```
```

File: packages/client/tests/e2e/prisma-client-error-message-win32/readme.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#Prisma Client Test for Windows Environment
+
+```
+Tests were created to check if the prisma client is installed properly for the Windows environment.
+For the Windows environments, the prisma client is unable to resolve packages and paths automatically compared to other environments when using Yarn. This results in a "Could not resolve @prisma/client..." error. Meaning that the client is unable to find the @prisma/client or resolve the path to or packages for the @prisma/client.
+
+Thus, Windows users must manually check if both the packages and @prisma/client are installed properly in order to successfully run the client.
```
Comment (Jolg42) [inline packages/client/tests/e2e/prisma-client-error-message-win32/readme.md:7]:
```suggestion
Tests were created to check if Prisma Client is installed properly on Windows.
On Windows, Prisma Client is unable to resolve packages and paths automatically compared to other environments when using Yarn. This results in a "Could not resolve @prisma/client..." error. Meaning that the client is unable to find the `@prisma/client` or resolve the path to or packages for `@prisma/client`.

Thus, Windows users must manually check if both the packages and `@prisma/client` are installed properly in order to successfully run the client.
```
Small changes

File: packages/client/tests/e2e/prisma-client-error-message-win32/readme.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#Prisma Client Test for Windows Environment
```
Comment (Jolg42) [inline packages/client/tests/e2e/prisma-client-error-message-win32/readme.md:1]:
```suggestion
# Prisma Client Test on Windows
```

**Comment by Jolg42** [comment]
(I still need to review the test)

**Comment by Templer44** [comment]
Updated the tests to show all tests passing and removed comments from the catchObjPromise test file. I've also removed the changes made to the prismaClientResolver file.

**Comment by jkomyno** [comment]
Hi @Templer44, I know it's been a while, but may I please ask you to update your PR to fix linting tests? (See [CI logs](https://github.com/prisma/prisma/actions/runs/10040748519/job/27752982550?pr=20604)).

Thanks!

**Comment by jkomyno** [comment]
Closing due to inactivity.

**Pull request closed without merging**